### PR TITLE
Support aliases

### PR DESCRIPTION
**Pull request by americk0**
# Description

We didn't have this work planned but I've been wanting to do this for some time now. This adds support for passing aliases anywhere that you can pass an account ID or account string. For example, before when calling `alks sessions open` you could pass the `-a/--account` and `-r/--role` and the value of the account field could be any of these options:

- 012345678910
- 012345678910/ALKSAdmin
- 012345678910/ALKSAdmin - awstest

but now you can also pass just the alias (`awstest`), so you don't have to memorize account IDs anymore. To do this I added a function that looks up account info based on whatever you passed to get the account ID, alias, and label from ALKS. This added an extra call to get all accounts that a user has via the `getAlksAccounts` function, which slowed performance, so I added some very short-lived caching to that function call to speed things up since it was getting called multiple times per command

#### Rally # [US852948](https://rally1.rallydev.com/#/?detail=/userstory/640447758755&fdp=true): ALKS CLI - Accept alias in place of account ID

# How has this been tested?

More unit tests were added to cover just about all the code that I touched, and I spent a while testing the commands manually to ensure that they still functioned correctly

### CODE REVIEW

File: src/lib/getAwsAccountFromString.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  if (alksAccount) {
+    return {
+      id: alksAccount.account.substring(0, 12),
+      alias: alksAccount.account.split(' - ')[1],
+      label: alksAccount.skypieaAccount?.label,
+    };
+  }
+
+  // if no matches or too many matches were found
```
Comment (elliottzack429) [inline src/lib/getAwsAccountFromString.ts:52]: I dont see here where you are confirming that there was only one match.  Looks like you just took the first match regardless of how many there were.  Either that logic should change or this comment should change to reflect that, unless I missed something.

**Review by elliottzack429 — CHANGES_REQUESTED** [review]
In addition to my inline comment, is it possible or reasonable to add examples in program.ts to the -a flag that shows a list of example account inputs when a user does -h, similar to how we do it for tags.

File: src/lib/getAwsAccountFromString.ts (hunk 1 above)
Comment (americk0) [reply src/lib/getAwsAccountFromString.ts:52]: Oh sorry that was from an early attempt at this. It's actually not a problem if multiple matches are found since "ALKS Account" actually means a pairing of AWS Account and IAM Role so it's common to have multiple ALKS Accounts for the same AWS Account. For example, I have the Admin and Security role for ALKS - Non Prod, so that would mean that if I passed `awsalksnp` as the alias then this would find 2 matches: `336692926357/ALKSAdmin` and `336692926357/ALKSSecurity`. 

Basically I realized that multiple matches are ok but forgot to remove this comment

**Comment by americk0** [comment]
> In addition to my inline comment, is it possible or reasonable to add examples in program.ts to the -a flag that shows a list of example account inputs when a user does -h, similar to how we do it for tags.

Yeah that makes sense. I have another PR coming to expand this to IAM-related commands but I think I'll just merge the two into this PR since I think I want them to release together. I'll get them merged and make this change

**Review by elliottzack429 — DISMISSED** [review]

**Review by elliottzack429 — APPROVED** [review]
